cmd/wsh/cmd/wshcmd-file.go

```
@@ -594,38 +498,32 @@ func fileListRun(cmd *cobra.Command, args []string) error {
-			if f.err != nil {
-				return f.err
+		for respUnion := range filesChan {
+			if respUnion.Error != nil {
+				log.Printf("error: %v", respUnion.Error)
+				return respUnion.Error
+			}
+			for _, f := range respUnion.Response.FileInfo {
+				fmt.Fprintln(os.Stdout, f.Name)
 			}
-			fmt.Fprintln(os.Stdout, f.info.Name)
+			return nil
```

golangci-lint (1.62.2)

SA4004: the surrounding loop is unconditionally terminated

(staticcheck)